core/opendaq/modulemanager/src/module_manager_impl.cpp

```diff
@@ -1173,6 +1195,36 @@ std::string ModuleManagerImpl::getPrefixFromConnectionString(std::string connect
     return "";
 }
 
+std::pair<std::string, tsl::ordered_map<std::string, BaseObjectPtr>> ModuleManagerImpl::splitConnectionStringAndOptions(
+    const std::string& connectionString)
+{
+    std::vector<std::string> strs1;
+    boost::split(strs1, connectionString, boost::is_any_of("?"));
+
+    if (strs1.size() == 1)
+        return std::make_pair(strs1[0], tsl::ordered_map<std::string, BaseObjectPtr> {});
+
+    if (strs1.size() != 2)
+        throw InvalidParameterException("Invalid connection string");
+
+    std::vector<std::string> options;
+    boost::split(options, strs1[1], boost::is_any_of("&"));
+
+    tsl::ordered_map<std::string, BaseObjectPtr> optionsMap;
+    for (const auto& option: options)
+    {
+        std::vector<std::string> keyAndValue;
+        boost::split(keyAndValue, option, boost::is_any_of("="));
+        if (keyAndValue.size() != 2)
+            throw InvalidParameterException("Invalid connection string");
+
+        BaseObjectPtr value = keyAndValue[1];
```

Comment:
Should we be using `EvalValue(keyAndValue[1])` here? Since now we're always calling `setPropertyValue()` with a string value, relying that the prop object implementation will properly convert the string.

Reply:
Good point, I will update the PR